cmd/junod/genica.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/cobra"
+
+	icacontrollertypes "github.com/cosmos/ibc-go/v3/modules/apps/27-interchain-accounts/controller/types"
+	icahosttypes "github.com/cosmos/ibc-go/v3/modules/apps/27-interchain-accounts/host/types"
+	icatypes "github.com/cosmos/ibc-go/v3/modules/apps/27-interchain-accounts/types"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/cosmos/cosmos-sdk/x/genutil"
+	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+)
+
+// AddGenesisAccountCmd returns add-genesis-account cobra Command.
+func AddGenesisIcaCmd(defaultNodeHome string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "add-ica-config",
+		Short: "Add ICA config to genesis.json",
+		Long:  `Add default ICA configuration to genesis.json`,
+		Args:  cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			serverCtx := server.GetServerContextFromCmd(cmd)
+			config := serverCtx.Config
+
+			config.SetRoot(clientCtx.HomeDir)
+
+			genFile := config.GenesisFile()
+			appState, genDoc, err := genutiltypes.GenesisStateFromGenFile(genFile)
+			if err != nil {
+				return fmt.Errorf("failed to unmarshal genesis state: %w", err)
+			}
+
+			controllerGenesisState := icatypes.DefaultControllerGenesis()
+			// no params set in upgrade handler, no params set here
+			controllerGenesisState.Params = icacontrollertypes.Params{}
+
+			hostGenesisState := icatypes.DefaultHostGenesis()
+			// add the messages we want
+			hostGenesisState.Params = icahosttypes.Params{
+				HostEnabled: true,
+				AllowMessages: []string{
+					"/cosmos.bank.v1beta1.MsgSend",
```

Comment:
Could we add `/cosmos/staking.v1beta1.MsgUndelegate` here?

Reply:
For reference, these are the `allow_messages` on the cosmos hub.
```
allow_messages:
- /cosmos.authz.v1beta1.MsgExec
- /cosmos.authz.v1beta1.MsgGrant
- /cosmos.authz.v1beta1.MsgRevoke
- /cosmos.bank.v1beta1.MsgSend
- /cosmos.bank.v1beta1.MsgMultiSend
- /cosmos.distribution.v1beta1.MsgSetWithdrawAddress
- /cosmos.distribution.v1beta1.MsgWithdrawValidatorCommission
- /cosmos.distribution.v1beta1.MsgFundCommunityPool
- /cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward
- /cosmos.feegrant.v1beta1.MsgGrantAllowance
- /cosmos.feegrant.v1beta1.MsgRevokeAllowance
- /cosmos.gov.v1beta1.MsgVoteWeighted
- /cosmos.gov.v1beta1.MsgSubmitProposal
- /cosmos.gov.v1beta1.MsgDeposit
- /cosmos.gov.v1beta1.MsgVote
- /cosmos.staking.v1beta1.MsgEditValidator
- /cosmos.staking.v1beta1.MsgDelegate
- /cosmos.staking.v1beta1.MsgUndelegate
- /cosmos.staking.v1beta1.MsgBeginRedelegate
- /cosmos.staking.v1beta1.MsgCreateValidator
- /cosmos.vesting.v1beta1.MsgCreateVestingAccount
- /ibc.applications.transfer.v1.MsgTransfer
- /tendermint.liquidity.v1beta1.MsgCreatePool
- /tendermint.liquidity.v1beta1.MsgSwapWithinBatch
- /tendermint.liquidity.v1beta1.MsgDepositWithinBatch
- /tendermint.liquidity.v1beta1.MsgWithdrawWithinBatch
```

Adding `/ibc.applications.transfer.v1.MsgTransfer` would also be appreciated!

Reply:
Nevermind, I see it's been added in #290